controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -52,14 +72,366 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+	r.ctx = context.Background()
+	r.namespace = request.Namespace
+
+	r.storageConsumer = &ocsv1alpha1.StorageConsumer{}
+	r.storageConsumer.Name = request.Name
+	r.storageConsumer.Namespace = r.namespace
+
+	r.cephBlockPool = &rookCephv1.CephBlockPool{}
+	r.cephBlockPool.Name = fmt.Sprintf("%s-%s", "cephblockpool", r.storageConsumer.Name)
+	r.cephBlockPool.Namespace = r.namespace
+
+	r.cephClientRBDProvisioner = &rookCephv1.CephClient{}
+	r.cephClientRBDProvisioner.Name = fmt.Sprintf("%s-%s", "cephclient-rbd-provisioner", r.storageConsumer.Name)
+	r.cephClientRBDProvisioner.Namespace = r.namespace
+
+	r.cephClientRBDNode = &rookCephv1.CephClient{}
+	r.cephClientRBDNode.Name = fmt.Sprintf("%s-%s", "cephclient-rbd-node", r.storageConsumer.Name)
+	r.cephClientRBDNode.Namespace = r.namespace
+
+	r.cephClientHealthChecker = &rookCephv1.CephClient{}
+	r.cephClientHealthChecker.Name = fmt.Sprintf("%s-%s", "cephclient-health-checker", r.storageConsumer.Name)
+	r.cephClientHealthChecker.Namespace = r.namespace
+
+}
+
+func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error) {
+
+	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateConfiguring
+
+	r.storageConsumer.Status.CephResources = nil
+
+	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
+		if !contains(r.storageConsumer.GetFinalizers(), storageConsumerFinalizer) {
+			r.Log.Info("Finalizer not found for StorageConsumer. Adding finalizer.", "StorageConsumer", klog.KRef(r.storageConsumer.Namespace, r.storageConsumer.Name))
+			r.storageConsumer.ObjectMeta.Finalizers = append(r.storageConsumer.ObjectMeta.Finalizers, storageConsumerFinalizer)
+			if err := r.Client.Update(r.ctx, r.storageConsumer); err != nil {
+				r.Log.Info("Failed to update StorageConsumer with finalizer.", "StorageConsumer", klog.KRef(r.storageConsumer.Namespace, r.storageConsumer.Name))
+				return reconcile.Result{}, err
+			}
+		}
+
+		if err := r.reconcileCephBlockPool(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileCephClientRBDProvisioner(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileCephClientRBDNode(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileCephClientHealthChecker(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		var notReady bool
+		for index := range r.storageConsumer.Status.CephResources {
+			cephObject := &r.storageConsumer.Status.CephResources[index]
+			if cephObject.Phase != "Ready" {
+				notReady = true
+			}
+		}
+
+		if !notReady {
+			r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateReady
+			r.storageConsumer.Status.GrantedCapacity = r.storageConsumer.Spec.Capacity
+		}
+
+	} else {
+		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDeleting
+
+		var cephResourcesExists bool
+		if err := r.Client.Delete(r.ctx, r.cephBlockPool); err == nil {
+			cephResourcesExists = true
```

Comment:
This does not make sense, if you managed to issue the delete then the resource should be considered as a non-existing resource. Am I missing something?

Reply:
After issuing the Delete operation , it is not guaranteed that the resource deletion is complete. Once the resource is deleted, it triggers the reconciliation and the flow does not reach this condition block.

Reply:
The flow has been modified to fetch the existence of cephResources first